plugin_guide.md

```diff
@@ -0,0 +1,184 @@
+
+```sh
+run-bmi-server --name "mymodel.bmi.MyModelBmi" --port 55555 --debug
+```
+
+## 4. Put your model in a container and add it to eWaterCycle 📦
+
+### Containerizing your model
+To be able to share your model, and have it work as an easily installable ewatercycle plugin, you need to create a container for your model.
+
+A `Dockerfile` is available in this repository for the LeakyBucket model, and can be modified to fit your own model. Note that [Docker](https://www.docker.com/) is required for this.
+
+To build the container locally, do:
+
+```sh
+docker build . -t "leakybucket"
+```
+Where `-t "name"` sets the "tag". 
+
+To start the container do:
+
+```sh
+docker run leakybucket
+```
+
+This should start the container and print the GRPC server's port to the terminal.
+
+The container tag can then be used to create a (local) containerized model:
+
+```py
+class LocalContainerLeakyBucket(ContainerizedModel, LeakyBucketMixins):
+    bmi_image: ContainerImage = ContainerImage("leakybucket")
+```
+
+The LocalContainerLeakyBucket model is demonstrated in [notebook 04](notebooks/04_containerized_model.ipynb)
+
+### Sharing the container using a registry
+
+To allow others to use the containerized model, the container has to be uploaded to a registry.
+One such registry is the Github container registry (ghcr).
+
+To see how to get an access token and upload to the ghcr, see [their guide](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry).
+
+The leaky bucket image has been pushed to `ghcr.io/ewatercycle/leakybucket`, with a tag corresponding to the current version of this repository.
+
+Instead of using the local container name, the `ContainerImage` class is given the `ghcr.io` location:
+
+```py
+class LeakyBucket(ContainerizedModel, LeakyBucketMixins):
+    """The LeakyBucket eWaterCycle model, with the Container Registry docker image."""
+    bmi_image: ContainerImage = ContainerImage(
+        "ghcr.io/ewatercycle/leakybucket:0.0.1"
+    )
+```
+
+### Register the model as eWaterCycle model entrypoint (plugin)
+
+Finally, the model can be registered as a plugin so that eWaterCycle can find it.
+This is done in the `pyproject.toml` file:
+
+```toml
+# This registers the plugin such that it is discoverable by eWaterCycle
+[project.entry-points."ewatercycle.models"]
+LeakyBucket = "leakybucket.ewatercycle_models:LeakyBucket"
+```
+
+Here you would replace the leaky bucket names with the correct model and class name of your own model.
+
+Now you can do:
+
+```py
+from ewatercycle.models import LeakyBucket
+```
+
+And run the model in eWaterCycle! 🚀
+
+For a working example, see [notebook 05](notebooks/05_finished_ewatercycle_plugin.ipynb)
+
+## Further steps
+
+After finishing the previous steps, you can upload the finished package to pypi.org.
+This will allow others to install it into their eWaterCycle installation using (for example):
+```sh
+pip install ewatercycle-mymodel
```

Comment:
Another step would be to get your model listed on the [eWatercycle plugins page](https://ewatercycle.readthedocs.io/en/latest/plugins.html). By making a [Pull request](https://github.com/eWaterCycle/ewatercycle/edit/main/docs/plugins.rst)